ambry-protocol/src/main/java/com.github.ambry.protocol/AdminRequest.java

```diff
@@ -81,6 +77,7 @@ public long writeTo(WritableByteChannel channel) throws IOException {
     long written = 0;
     if (bufferToSend == null) {
       serializeIntoBuffer();
+      bufferToSend.flip();
```

Comment:
Because `RequestControlAdminRequest` implements its own version of `serializeIntoBuffer() ` and it calls `super.serializeIntoBuffer() ` before adding its own stuff. When it is adding its own stuff, the buffer should not already have been flipped.